### PR TITLE
Add language around supported number of versions for languages

### PR DESCRIPTION
**Pull request by ericdill**
This came up in the @conda-forge/core meeting today. We agreed that our current policy should be documented. Please have a read and make suggestions or direct pushes to the branch.

### CODE REVIEW

**Comment by isuruf** [comment]
I think this goes beyond what our current policy is. We didn't have this policy when dropping 2.7. (We supported 2.7 even after numpy, scipy dropped support). We should vote on it since this is a new policy.

**Comment by isuruf** [comment]
I looked at a few previous discussions.
- 3.4 was dropped because of CI constrains which we don't have now - https://github.com/conda-forge/conda-smithy/pull/421
- There was no discussion for dropping 3.5 in https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/127

**Comment by beckermr** [comment]
Thanks for the history lesson @isuruf !

Note that we did not vote when we merged the guidelines and principles we use to determine when compilers are upgraded. 

As nothing in these docs is binding but is more a statement of general principles we consider, I don't think we need a vote. A more specific policy would require a vote imho.

**Comment by scopatz** [comment]
This language looks really good to me!

File: src/orga/guidelines.rst [hunk 1]
```diff
@@ -136,3 +136,29 @@ Becoming a maintainer
...
+Each of these language-specific packaging sub-ecosystems needs to keep cadence with the language itself, making it challenging to have a blanket policy for how long to keep older versions of the language around.
+As it comes up, each group should be able to define their own policy on how long to keep older versions of their language around.
+
+Python
+------
+For the Python language, conda-forge aims to keep package builds active and available for the current version and the two preceding minor versions.
+Whenever Python 4.0 comes out we'll need to figure out if this policy should change to support multiple versions of 3.x and 4.x simultaneously. 
+Fortunately, we can punt on that for now.
+The question of when to decide to drop an older language version remains.
+The guidance that we can provide here is twofold:
```
Comment (isuruf) [inline src/orga/guidelines.rst:154]:
```suggestion
The guidance that we can provide here is two fold:
```

File: src/orga/guidelines.rst [hunk 2]
```diff
@@ -136,3 +136,29 @@ Becoming a maintainer
...
+
+1. We will move with the community. 
+   When our core libraries stop supporting an old version, so too will conda forge.
+   The (nonexhaustive) list of core libraries that we consider when making the decision to drop an older version are:
+   * matplotlib
+   * numpy
+   * scipy
+   * pypy
+2. The core team can decide to keep an old version around temporarily until some specific criteria is met.
+   For example, we're holding off on turning off py36 until pypy comes out with pypy3.7.
```
Comment (isuruf) [inline src/orga/guidelines.rst:164]:
```suggestion
   For example, we're holding off on turning off py36 until pypy comes out with pypy3.7.
3. If there are lots of people in the community relying on older versions, core team can decide to keep an old version around
   For example, we held off turning off py27 even after numpy, scipy dropped support as there were many in the community interested in keeping support until the end of life of that version.
```

File: src/orga/guidelines.rst [hunk 3]
```diff
@@ -136,3 +136,31 @@ Becoming a maintainer
...
+Language versions
+=================
+
+Conda-forge contains packages from a number of languages including Python and R, among many others.
+Each of these language-specific packaging sub-ecosystems needs to keep cadence with the language itself, making it challenging to have a blanket policy for how long to keep older versions of the language around.
+As it comes up, each group should be able to define their own policy on how long to keep older versions of their language around.
+
+Python
+------
+For the Python language, conda-forge aims to keep package builds active and available for the current version and the two preceding minor versions.
```
Comment (isuruf) [inline src/orga/guidelines.rst:150]:
```suggestion
For the Python language, conda-forge aims to keep package builds active and available for the current version and at least two preceding minor versions.
```

**Comment by beckermr** [comment]
any other comments @conda-forge/core?

**Comment by ericdill** [comment]
Thanks for the feedback all. I'm glad we got this sorted out and into the docs

**Comment by jakirkham** [comment]
Thanks Eric! 😄

**Pull request merged**